plaso/engine/engine.py

```diff
@@ -272,3 +278,42 @@ def SupportsGuppyMemoryProfiling(cls):
       bool: True if memory profiling with guppy is supported.
     """
     return profilers.GuppyMemoryProfiler.IsSupported()
+
+  @classmethod
+  def BuildFilterFindSpecs(
+      cls, artifacts_registry, knowledge_base_object,
+      artifact_filter_names=None, filter_file_path=None):
+    """Build Find Specs from artifacts or filter file if available.
+
+    Args:
+       knowledge_base_object (KnowledgeBase): Knowledge base.
+       artifact_filter_names (Optional list[str]): Artifact filter names.
+       filter_file_path (Optional [str]): Path of filter file.
+
+    Returns:
+      list[dfvfs.FindSpec]: find specifications for the file source type.
+
+    Raises:
+      RuntimeError: if no valid FindSpecs are built.
+    """
+
+    environment_variables = knowledge_base_object.GetEnvironmentVariables()
+    find_specs = None
+    if artifact_filter_names:
+      artifact_filters_object = (
+          artifact_filters.ArtifactDefinitionsFilterHelper(
+              artifacts_registry, artifact_filter_names, knowledge_base_object))
+      artifact_filters_object.BuildFindSpecs(environment_variables)
+      find_specs = knowledge_base_object.GetValue(
+          artifact_filters_object.KNOWLEDGE_BASE_VALUE)[
+              artifact_types.TYPE_INDICATOR_FILE]
+    elif filter_file_path:
+      filter_file_object = filter_file.FilterFile(filter_file_path)
+      find_specs = filter_file_object.BuildFindSpecs(
+          environment_variables=environment_variables)
+
+    if (artifact_filter_names or filter_file_path) and not find_specs:
+      raise RuntimeError('Error processing filters, no valid specifications'
```

Comment:
I opt to break after `(` for consistency

Reply:
Done.